scripts/py_matter_yamltests/generate_yaml_doc_tables.py

```diff
@@ -0,0 +1,70 @@
+#
+#    Copyright (c) 2023 Project CHIP Authors
+#
+#    Licensed under the Apache License, Version 2.0 (the 'License');
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an 'AS IS' BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+import os
+from typing import TextIO
+
+from matter_yamltests.yaml_loader import SchemaTree, yaml_tree
+
+SCRIPT_DIR = os.path.abspath(os.path.dirname(__file__))
+WARNING = ("<!---\n"
+           "This file is automatically generated by a script.\n"
+           "DO NOT HAND-EDIT THIS FILE.\n"
+           f"Script: {os.path.basename(__file__)}\n"
+           "-->\n\n")
+
+
+def get_type_list_and_vars(typetuple) -> (list[type], bool):
+    # If str is one of the supported types, and other base types are supported,
+    # this means it supports variables.
+    # This is a heuristic, but it's true for now.
+    try:
+        typelist = list(typetuple)
+        if str in typelist:
+            reduced = [t for t in typelist if t != str]
```

Comment:
yeah, it's a comparison to a class. It's not an instance though - it's the class of the type, which is why it looks weird. ie, the yaml parser has a tuple of classes that are allowed. So t will directly be str class, rather than an instance of the class or a string of the class name. That happens because the parser is coded weirdly. I've added a comment to explain.